storm-core/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -0,0 +1,156 @@
+import org.apache.storm.cluster.DaemonType;
+import org.apache.storm.metrics2.reporters.StormReporter;
+import org.apache.storm.task.WorkerTopologyContext;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+public class StormMetricRegistry {
+
+    private static final Logger LOG = LoggerFactory.getLogger(StormMetricRegistry.class);
+
+    private static final MetricRegistry REGISTRY = new MetricRegistry();
+
+    private static final List<StormReporter> REPORTERS = new ArrayList<>();
+
+    private static String hostName = null;
+
+    public static <T> SimpleGauge<T>  gauge(T initialValue, String name, String topologyId, String componentId, Integer port){
+        SimpleGauge<T> gauge = new SimpleGauge<>(initialValue);
+        String metricName = metricName(name, topologyId, componentId, port);
+        if(REGISTRY.getGauges().containsKey(metricName)){
+            return (SimpleGauge)REGISTRY.getGauges().get(metricName);
+        } else {
+            return REGISTRY.register(metricName, gauge);
+        }
+    }
+
+    public static DisruptorMetrics disruptorMetrics(String name, String topologyId, String componentId, Integer port){
+        return new DisruptorMetrics(
+                StormMetricRegistry.gauge(0L, name + "-capacity", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-population", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-write-position", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-read-position", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0.0, name + "-arrival-rate", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0.0, name + "-sojourn-time-ms", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-overflow", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0.0F, name + "-percent-full", topologyId, componentId, port)
+        );
+    }
+
+    public static Meter meter(String name, WorkerTopologyContext context, String componentId, String executorId, String streamId){
+        String metricName = metricName(name, context.getStormId(), componentId, streamId,executorId, context.getThisWorkerPort());
+        return REGISTRY.meter(metricName);
+    }
+
+    public static Counter counter(String name, WorkerTopologyContext context, String componentId, String executorId, String streamId){
+        String metricName = metricName(name, context.getStormId(), componentId, streamId,executorId, context.getThisWorkerPort());
+        return REGISTRY.counter(metricName);
+    }
+
+    public static void start(Map<String, Object> stormConfig, DaemonType type){
+        String localHost = "localhost";
```

Comment:
I think this line is unused, or perhaps you actually wanted to default the hostName to "localhost" but ended up declaring a variable instead.